### PR TITLE
fix(SessionTracker): remove automatic session capturing for Lumen 5.3+

### PR DESCRIPTION
**Pull request by tomlongridge**
## Goal

`session()` is not available after Lumen 5.3 - this change prevents session tracking setup if the
method is not available.

### Changed

- BugsnagServiceProvider: setupSessionTracking now does nothing if `session` is not available.

## Tests

Manually tested that no exception occurs on Lumen and that session information continues to work on Laravel.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: src/BugsnagServiceProvider.php [hunk 1]
```diff
@@ -372,6 +372,12 @@ protected function setupPaths(Client $client, $base, $path, $strip, $project)
      */
     protected function setupSessionTracking(Client $client, $endpoint, $events)
     {
+        // Session support removed in Lumen 5.3 - only setup automatic session
+        // tracking if the session function is avaiable
+        if (!function_exists('session')) {
+            return;
```
Comment (Cawllec) [inline src/BugsnagServiceProvider.php]: I think it would be worth adding a log here to mention that sessions are not automatically being tracked, as someone may expect if they set the configuration option.

Comment (tomlongridge) [reply src/BugsnagServiceProvider.php]: I'm struggling to log anything (using Laravel logging) as it calls recursively into the Bugsnag setup code for the BugsnagLogger handling. Is there a lower-level approach that could be used to log?

Comment (tomlongridge) [reply src/BugsnagServiceProvider.php]: Also, given the fact that it doesn't currently work and there should be an alternative solution to allow support for this in the not-too-distant-future, I think it's not too important to log this if it proves difficult.

Comment (Cawllec) [reply src/BugsnagServiceProvider.php]: Yeah, I hadn't considered the logging straight to Bugsnag aspect.  In theory it shouldn't be a problem if the log-level is high enough, but there's no guarantee there, and I haven't actually tried it myself.

In which case it's probably fine to not log.
